services/diversity/src/services/richness.js

```diff
@@ -12,105 +12,109 @@ module.exports = (RichnessPersistence, restAPI) => {
      *
      * @returns {Object[]} Number of inferred and observed species for the desired group.
      */
-    getNumberSpecies: async (areaType, areaId, group = 'all') => {
-      const data = [
-        {
-          id: 'total',
-          inferred: 40,
-          observed: 4,
-          region_observed: 110,
-          region_inferred: 130,
-        },
-        {
-          id: 'endemic',
-          inferred: 20,
-          observed: 25,
-          region_observed: 120,
-          region_inferred: 100,
-        },
-        {
-          id: 'invasive',
-          inferred: 20,
-          observed: 10,
-          region_observed: 90,
-          region_inferred: 80,
-        },
-        {
-          id: 'threatened',
-          inferred: 15,
-          observed: 20,
-          region_observed: 90,
-          region_inferred: 100,
-        },
-      ];
+    getNumberOfSpecies: async (areaType, areaId, group = 'all') => {
+      const promises = [];
+      let result;
       switch (group) {
         case 'total':
-          return [data[0]];
+          promises.unshift(RichnessPersistence.findTotalNumberOfSpecies(areaType, areaId));
+          break;
         case 'endemic':
-          return [data[1]];
+          promises.unshift(RichnessPersistence.findEndemicNumberOfSpecies(areaType, areaId));
+          break;
         case 'invasive':
-          return [data[2]];
+          promises.unshift(RichnessPersistence.findInvasiveNumberOfSpecies(areaType, areaId));
+          break;
         case 'threatened':
-          return [data[3]];
+          promises.unshift(RichnessPersistence.findThreatenedNumberOfSpecies(areaType, areaId));
+          break;
         case 'all':
+          promises.unshift(
+            RichnessPersistence.findTotalNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findEndemicNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findInvasiveNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findThreatenedNumberOfSpecies(areaType, areaId),
+          );
+          break;
         default:
-          return data;
+          return [];
```

Comment:
I think it's better to return an error, in theory this case never executes, but it's clearer that way, what do you think @cazapatamar ?

Reply:
Yes, it's better to return the error so we can differentiate the case when actually there is no data.

Reply:
@erikasv and @cazapatamar No matter what is returned in default case, the error is not thrown, the item was always returned in lines 46, 52, 110 and 116 (richness.js) then I changed these lines to do not return item with null values and the default case is still returning empty array. Is it ok?

Reply:
The default case could be reached if parameter `group` does not correspond to any of the options (all, total, etc). Thinking of the appropriate endpoint response in this case, it is better to return the specific error instead of an empty array which could be confusing when actually there is no data in the database.

I mean, if we return an empty array the user (that one who access the endpoint) may think there is not data instead of realize there is a problem with the endpoint params.

Reply:
Yeah, agree with @cazapatamar

Reply:
I added error in default case.